### PR TITLE
tools: update requirements for nrf-regtool

### PR DESCRIPTION
**Pull request by PerMac**
5.6.0 is required by
https://github.com/zephyrproject-rtos/zephyr/
blob/main/modules/hal_nordic/CMakeLists.txt#L15.

### CODE REVIEW

**Review by gmarull — CHANGES_REQUESTED** [review]
this requires bringing pinctrl clockpin changes

**Comment by shanthanordic** [comment]
@57300  should approve this PR before merging.

**Comment by PerMac** [comment]
> this requires bringing pinctrl clockpin changes

I assume those will come with the ongoing upmerge? Without newer regtool nrf54h20 prints no output. We can get it installed within our CI for now to unlock testing.

**Comment by gmarull** [comment]
> > this requires bringing pinctrl clockpin changes
> 
> I assume those will come with the ongoing upmerge? Without newer regtool nrf54h20 prints no output. We can get it installed within our CI for now to unlock testing.

I guess this is upstream only, not downstream. They'll come with the next upmerge, if not, they can be easily cherry-picked in advance (it's strictly this single commit: 5e3188605ed94b3d2a01647187324415e31f4d07).

**Review by 57300 — CHANGES_REQUESTED** [review]
The required pinctrl changes are expected to go in with https://github.com/nrfconnect/sdk-zephyr/pull/1952

**Comment by 57300** [comment]
Updated as part of #17197

**Pull request closed without merging**
